core/components/versionx/controllers/index.php

```diff
@@ -37,12 +37,16 @@
             $v = $versionx->getVersionDetails('vxResource',$versionid,true);
             if ($v !== false)
                 $modx->regClientStartupHTMLBlock('<script type="text/javascript">VersionX.record = '.$v.'; </script>');
+            else
+                return $modx->error->failure($modx->lexicon('versionx.error.noresults'));
```

Comment:
The logic is maybe a little weird in the controller, but doesn't the javascript already catch the record not being available?

Reply:
The user will end up with a blank MODX manager page and a javascript error will be present for the missing VersionX.record. The same will be true when you change the request to a non-existing id. Eg: manager/?a=4&action=resource&vid=10000. Catching the error in javascript would in my opinion be the better solution if the version was actually received through a ajax request. 

Since this is not the case it would be resulting in (with the HTTP status codes in mind) a 403 forbidden request. So in this case I think reporting a error (through modError) makes sense. The modManagerResponse is formatting the error later on in the request, just like trying to edit a modResource without access to the instance:
````php
    /**
     * Send the response to the client
     */
    public function send() {
        if (is_array($this->body)) {
            $this->modx->smarty->assign('_e', $this->body);
            if (!file_exists($this->modx->smarty->template_dir.'error.tpl')) {
                $templatePath = $this->modx->getOption('manager_path') . 'templates/default/';
                $this->modx->smarty->setTemplatePath($templatePath);
            }
            echo $this->modx->smarty->fetch('error.tpl');
        } else {
            echo $this->body;
        }
        @session_write_close();
        exit();
    }
````
https://github.com/modxcms/revolution/blob/2ad1f763a32afd28fd5b89e7f0ba5ceaa26bfad1/core/model/modx/modmanagerresponse.class.php#L120

Reply:
Not a proper message being sent is handled client-side now (#106) as of 2.2.2